src/test/scala/chiselTests/experimental/ProgrammaticPortsSpec.scala

```diff
@@ -63,11 +66,12 @@ class ProgrammaticPortsSpec extends ChiselFlatSpec with Utils {
   }
 
   "SuggestName collisions on ports" should "be illegal" in {
-    a[ChiselException] should be thrownBy extractCause[ChiselException] {
+    val error = intercept[Exception] {
       ChiselStage.elaborate(new Module {
         val foo = IO(UInt(8.W)).suggestName("apple")
         val bar = IO(UInt(8.W)).suggestName("apple")
       })
     }
+    assert(error.toString contains "Cannot have duplicate names apple_1 and apple")
```

Comment:
would there be any way to reflect produce the source locator of the things with these names?

Reply:
It would be wonderful, but we would need to capture and keep source locators upon naming and that's a bit tricky. It's worth considering in future enhancements though.